src/payment/strategies/bluesnapv2/bluesnapv2-payment-strategy.ts

```diff
@@ -0,0 +1,68 @@
+import { CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { OrderFinalizationNotRequiredError } from '../../../order/errors';
+import { PaymentArgumentInvalidError } from '../../errors';
+import PaymentActionCreator from '../../payment-action-creator';
+import { PaymentInitializeOptions, PaymentRequestOptions } from '../../payment-request-options';
+import * as paymentStatusTypes from '../../payment-status-types';
+import PaymentStrategy from '../payment-strategy';
+
+import BlueSnapV2PaymentPageLoader from './bluesnapv2-payment-page-loader';
+
+export default class BlueSnapV2PaymentStrategy implements PaymentStrategy {
+
+    private readonly TARGET = 'bluesnapv2_hosted_payment_page';
+
+    constructor(
+        private _store: CheckoutStore,
+        private _orderActionCreator: OrderActionCreator,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _blueSnapV2PaymentPageLoader: BlueSnapV2PaymentPageLoader
+    ) {}
+
+    execute(orderRequest: OrderRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        const { payment } = orderRequest;
+
+        if (!payment) {
+            throw new PaymentArgumentInvalidError(['payment']);
+        }
+
+        return this._store.dispatch(
+            this._orderActionCreator.submitOrder(orderRequest, options)
+        )
+        .then(() =>
+            Promise.race([
+                this._blueSnapV2PaymentPageLoader.loadPaymentPage(this.TARGET),
+                this._store.dispatch(
+                    this._paymentActionCreator.initializeOffsitePayment(payment.methodId, payment.gatewayId, this.TARGET),
+                    {
+                        queueId: 'BlueSnapV2PaymentStrategy:initializeOffsitePayment:' + Math.random().toString(36).slice(-7),
```

Comment:
what do you need a random queueId for?

Reply:
I'm not sure how `store.dispatch()` works but, if I close the modal and open it again, initializeOffsitePayment keeps the last resolved state (rejected) and I'm not able to execute it again.

Reply:
Sorry, let me rewrite my explanation again... The random queueId is related to my previous comment: https://github.com/bigcommerce/checkout-sdk-js/pull/732#discussion_r354602920. If I don't do this, I'm unable to dispatch initializeOffsitePayment again.